src/NUnitFramework/framework/Internal/MethodWrapper.cs

```diff
@@ -165,15 +170,23 @@ public IMethodInfo MakeGenericMethod(params Type[] typeArguments)
         /// </summary>
         public T[] GetCustomAttributes<T>(bool inherit) where T : class
         {
+#if NETCORE
+            return MethodInfo.GetAttributes<T>(inherit).ToArray();
+#else
             return (T[])MethodInfo.GetCustomAttributes(typeof(T), inherit);
+#endif
         }
 
-        /// <summary>
-        /// Gets a value indicating whether one or more attributes of the spcified type are defined on the method.
-        /// </summary>
+            /// <summary>
+            /// Gets a value indicating whether one or more attributes of the spcified type are defined on the method.
+            /// </summary>
         public bool IsDefined<T>(bool inherit)
         {
+#if NETCORE
+            return MethodInfo.GetCustomAttributes(inherit).Any(a => typeof(T).IsAssignableFrom(a.GetType()));
```

Comment:
IsDefined also throws if T does not derive from Attribute, thus the custom implementation.